docs/docs/01-ibc/10-roadmap.md

```diff
@@ -7,35 +7,166 @@ slug: /ibc/roadmap
+- Implementation of IBC on new platforms and environments is difficult and resource intensive.
+- IBC's multiple layers of abstraction are burdensome, confusing, and unnecessary.
+- IBC demands lots of state to be written and messages exchanged (for connection and channel handshakes) before the first user-level message can be send sent (e.g. before a transfer packet can be sent).
+- Upgradability is difficult and slow, making innovation on IBC difficult and slow.
+
+With the code-named project Eureka we want to address these problems by vastly simplifying the protocol, but without compromising or reducing security and functionality. At a high level, IBC Eureka:
+
+- Removes connection and channel handshakes.
+- Retains interoperability using a light client based security model.
+- Retains existing packet semantics (send, receive, acknowledge, timeout).
+- Uses a single light client and channel for all applications.
+
+For more details, please check [#6985](https://github.com/cosmos/ibc-go/issues/6985). The initial Eureka design was proposed in [cosmos/ibc#1093](https://github.com/cosmos/ibc/pull/1093).
+
+### Compatibility
+
+In v10 there will be two implementations of the IBC protocal living side by side. We will call them classic (or v1) and eureka (or v2). The classic implementation remains completely backwards compatible with all previous versions of ibc-go (i.e. connection and channel handshakes, channel upgradability and packet processing will work as they do today), but the eureka implementation will be compatible only with other IBC Eureka implementations (e.g. [solidity-ibc-eureka](https://github.com/cosmos/solidity-ibc-eureka) for EVMs) and other chains running ibc-go v10.
+
+Chains running ibc-go v10 have the choice then to communicate between them using either classic or eureka implementations. For example: given that chain A (with ibc-go v10) already had a transfer channel set up with chain B (also running ibc-go v10), if tokens need to be sent from chain A to chain B, then that can be done using the classic implementation or the eureka implementation. Most likely the indication of which implementation to use will be in a new `protocol_version` field in `MsgSendPacket`:
+
+```proto
+message MsgSendPacket {
+  string                    source_port         = 1;
+  string                    source_channel      = 2;
++ string                    destination_port    = 3;
++ string                    destination_channel = 4;
```

Comment:
Not sure if we will need this, but added it for now (if we use channel aliasing, we should be able to find the counterparty channel ID; and if we use a client ID, we should be able also to get the counterparty client ID that was stored with `MsgProvideCounterparty`).